src/Rules/Program.hs

```diff
@@ -18,9 +17,9 @@ import Utilities
 buildProgram :: [(Resource, Int)] -> Rules ()
 buildProgram rs = do
     root <- buildRootRules
-    forM_ [Stage0 ..] $ \stage ->
-        [ root -/- stageString stage -/- "bin"     -/- "*"
-        , root -/- stageString stage -/- "lib/bin" -/- "*" ] |%> \bin -> do
+    forM_ [Stage0 .. Stage2] $ \stage ->
+        [ root -/- stageString (succ stage) -/- "bin"     -/- "*"
+        , root -/- stageString (succ stage) -/- "lib/bin" -/- "*" ] |%> \bin -> do
```

Comment:
It would probably be clearer to just loop over the stages that we actually care about, wouldn't it?

Reply:
No, `stage` always corresponds to the current build stage. The `succ` here is merely to point to the right build directory.